compiler/arm/codegen/ControlFlowEvaluator.cpp

```diff
@@ -81,7 +81,14 @@ TR::Register *OMR::ARM::TreeEvaluator::ireturnEvaluator(TR::Node *node, TR::Code
    addDependency(deps, returnRegister, cg->getProperties().getIntegerReturnRegister(), TR_GPR, cg);
 
    generateAdminInstruction(cg, ARMOp_ret, node, deps);
-   cg->comp()->setReturnInfo(TR_IntReturn);
+   if (node->getDataType() == TR::Int8)
```

Comment:
This would be much more elegant as a switch on node->getDataType() with unsupported types hitting a fatal assertion with an informative message.

Reply:
...particularly since there's probably a follow-on PR to add support for sreturn coming, I would guess :)